File: leetcode.com/1752_check_if_array_is_sorted_and_rotated/solution.c
bool check(int* nums, int numsSize)
{
    int transitions = 0;

    for (int i = 0 ; i < numsSize - 1 ; i++) {
        if (nums[i] > nums[i + 1])
            transitions++;
    }

    if (nums[numsSize - 1] > nums[0])
        transitions++;

    return (transitions > 1) ? false : true;
}

File: leetcode.com/1752_check_if_array_is_sorted_and_rotated/solution.go
func check(nums []int) bool {
    bp := 0

    for i := 0 ; i < len(nums) ; i++ {
        if nums[i] > nums[(i + 1) % len(nums)] {
            bp++
        }
    }

    if bp > 1 {
        return false
    }

    return true
}